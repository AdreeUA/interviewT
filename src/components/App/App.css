.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  padding: 20px;
}

.App-header > h1 {
  color: white;
}

.App-intro {
  font-size: large;
}

.App-nav a {
  display: inline-block;
  vertical-align: top;
  color: #ffffff;
  margin: 0 15px;
  padding: 5px 10px;
  border: 1px solid #ccc;
  text-decoration: none;
  transition: .5s;
}

.App-nav a:hover {
  color: #222;
  background-color: #fff;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
