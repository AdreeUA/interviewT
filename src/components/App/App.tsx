import React, { Component } from 'react'
import { renderRoutes } from 'react-router-config'
import './App.css'

const logo = require('../../resources/logo.svg')

class App extends Component<{ route: any }, {}> {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        {renderRoutes(this.props.route.routes)}
      </div>
    )
  }
}

export default App
