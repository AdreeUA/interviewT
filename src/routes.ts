import App from './components/App'
import Posts from './components/Posts'
import Post from './components/Post'
import Users from './components/Users'

export default [
  {
    component: App,
    routes: [
      {
        path: '/',
        exact: true,
        component: Posts,
      },
      {
        path: '/posts/:id',
        component: Post,
      },
      {
        path: '/users',
        component: Users,
      },
    ],
  },
]
