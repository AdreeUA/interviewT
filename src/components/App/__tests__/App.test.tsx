import * as React from 'react'
import { shallow } from 'enzyme'
import routes from '../../../routes'
import App from '../App'

describe('App', () => {
  it('render()', () => {
    const route = routes[0]
    const wrapper = shallow(<App route={route} />)
    expect(wrapper).toMatchSnapshot()
  })
})
