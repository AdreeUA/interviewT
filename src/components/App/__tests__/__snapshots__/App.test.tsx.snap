// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render() 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="test-file-stub"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <Switch>
    <Route
      exact={true}
      key="0"
      path="/posts"
      render={[Function]}
    />
    <Route
      key="1"
      path="/posts/:id"
      render={[Function]}
    />
    <Route
      key="2"
      path="/"
      render={[Function]}
    />
  </Switch>
</div>
`;
